test/jdk/java/lang/Thread/virtual/TraceVirtualThreadLocals.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @summary Test diagnostic option for detecting a virtual thread using thread locals
+ * @library /test/lib
+ * @run junit/othervm -Djdk.traceVirtualThreadLocals TraceVirtualThreadLocals
+ * @run junit/othervm -Djdk.traceVirtualThreadLocals=true TraceVirtualThreadLocals
+ */
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import jdk.test.lib.thread.VThreadRunner;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+class TraceVirtualThreadLocals {
+
+    @Test
+    void testInitialValue() throws Exception {
+        String output = run(() -> {
+            ThreadLocal<String> name = new ThreadLocal<>() {
+                @Override
+                protected String initialValue() {
+                    return "<unnamed>";
+                }
+            };
+            name.get();
+        });
+        assertContains(output, "java.lang.ThreadLocal.get");
```

Comment:
Should it also assert the presence of `ThreadLocal.setInitialValue` in the stacktrace?

Reply:
It's the `get` method that triggers the initial value to be generated so this is why the test checks that method. But I think you have a point it would be clearer if the the test were to match on something that has "initialValue" in the string. It means using the name of an internal method but it might be okay here.